### ISSUE TITLE
Getting node id from TasksService

### ISSUE THREAD

**Issue by BArtistically** [nathanwoulfe/Plumber-2#30]
Hi Nathan,
I was wondering how to get the node id of the document when using the TasksService
```
TasksService.Created += TasksService_Created;

private void TasksService_Created(object sender, TaskEventArgs e)
        {
            var task = e.Task;
        }
```
The WorkflowInstance is null and via the e.Task.WorkflowInstanceGuid property I can't look it up in the database the first 
time because it doesn't exists. Shouldn't it be available?

Is it also possible to clean cancel the 'approve' as we need to not publish it yet but set a status to 'ready for publication'

I tried it in ContentService.Publishing but when I cancel it always gives me an workflow error.

Is it possible? 
Thank you in advance!

**Comment by nathanwoulfe** [comment]
Hi @BArtistically 

Looks like what's happening is the Created event is raised when the new Task is persisted - which can happen before its related WorkflowInstance is persisted (in the case of a new workflow, the Instance is created, the first Task is created and persisted, the Instance is updated and finally persisted). On an existing workflow you'd be able to get the Instance from the Guid because the Instance will always exist in the DB. That said, I can add property to the Task to provide the nodeId, which would avoid any additional trips to the database.

Looking at the code, I think I started cleaning up events to shift them to more sensible locations (ie raised after objects are persisted) but never finished it. Easy enough to do though, will get back on it.

As for cancelling the publish, it's not possible directly via Plumber, but I'd have expected hooking the Umbraco Publishing event would work fine - by the time we get to publishing, it's all native Umbraco functionality. Although, looking at the code now, there's an UmbracoOperationFailedException raised if the response from ContentService.SaveAndPublish is !Success, so that's possibly the error you're seeing? I think that's some legacy stuff from v7, related to managing notifications. If you could share a log/trace, that would be a big help.

**Comment by BArtistically** [comment]
Hi Nathan,

Thank you for responding so fast. Yeah, it would be nice to have the workflowinstance filled in on the created task.
For the cancelling, sorry, i don't have a log/trace but that's exact what you described. It gives a UmbracoOperationFailedException and the screen doesn't get updated anymore so I was stuck there. 
For now, i had to make a workaround not using Plumber for the 'ready for publication' feature. 
But maybe in a later stage of the project we will be using it again as it has a lot nice features we can use.

**Comment by nathanwoulfe** [comment]
@BArtistically I've published a patch (1.3.7, on NuGet now) which should resolve this - Plumber shouldn't raise an error when the publish is cancelled, only if the publish fails. Also adds a status for those workflow instances, to mark them as cancelled by a third party. NodeId is now available on the WorkflowInstance and WorkflowTaskInstance models, and event are raised at more appropriate times.
